File: java_project/oops/NAUGHTEEN/LeatherItems.java
package oops.NAUGHTEEN;

public abstract class LeatherItems {
    protected String itemId;
    protected String billId;
    protected String category;
    protected String type;

    public LeatherItems(String itemId, String category, String type ){
        this.itemId = itemId;
        this.category = category;
        this.type = type;
    }
    public String getItemId() {
        return itemId;
    }

    public String getBillId() {
        return billId;
    }

    public String getType() {
        return type;
    }

    public String getCategory() {
        return category;
    }
    public boolean validateCategory(){
        if (this.category!= null && (this.category.equalsIgnoreCase("Imported") || this.category.equalsIgnoreCase("Manufactured"))){
            return true;
        }
        return false;
    }
    public abstract boolean validateType();
    public abstract float identifyCost();
    public abstract float calculateBillAmount();
    
}

File: python_project/oops/NAUGHTEEN/LeatherItems.py
from abc import ABC, abstractmethod


class LeatherIteams(ABC):
    def __init__(self, item_id, category,item_type):
        self. item_id = item_id
        self. category = category
        self.item_type = item_type
        self.bill_id = None

    def validate_category(self):
        return self.category.lower()in ["imported", "manufactured"]
        
    @abstractmethod
    def validate_type(self):
        pass
    @abstractmethod
    def identify_cost(self):
        pass
    @abstractmethod
    def calculate_bill_amount(self):
        pass
        